src/coreclr/jit/utils.cpp

```diff
@@ -1629,7 +1629,6 @@ void HelperCallProperties::init()
             case CORINFO_HELP_READYTORUN_GENERIC_HANDLE:
                 // logging helpers are not technically pure but can be optimized away
                 isPure        = true;
-                noThrow       = true;
                 nonNullReturn = true;
                 break;
 
```

Comment:
```suggestion
                noThrow       = true;
                nonNullReturn = true;
                break;

            case CORINFO_HELP_READYTORUN_GENERIC_HANDLE:
                // Typically this is a no-throw call, but may become always-throw when fetching an unavailable handle.
                isPure        = true;
                nonNullReturn = true;
                break;
```
And take out `case CORINFO_HELP_READYTORUN_GENERIC_HANDLE` from above. There is no need to pessimize the other helpers.

Reply:
Done, thanks.